### PR TITLE
Implementation of a new EBL model from Finke et al. 2022.

### PR DESCRIPTION
**Pull request by Yannick-Schmidt**
Hi all,

this PR adds the recent EBL model from [Finke et al. 2022](https://arxiv.org/abs/2210.01157)&mdash;which is basically an improved version of the Finke10 model already implemented in CRPropa&mdash;to the existing selection of custom photon fields. As advised in the paper the corresponding data files (i.e. comoving photon density data for various redshift values) were extracted from [https://zenodo.org/records/7023073](https://zenodo.org/records/7023073).

The implementation includes the ...
- ... general creation of a new class (in the `photonField.py` script).
- ... adding of all necessary data files into the 'tables ' folder.
- ... models' adding to the 'fields_cmbebl'-list (`calc_all.py`).

The functionality of the new class(es) (with regard to the other implemented class in the CRPropa repo) has already been tested and works without any problems so far. For the purpose of a brief comparison the comoving photon densities of some selected EBL models, including the newly implemented one, are shown below.
EBL_comparison_withlim

### CODE REVIEW

File: photonField.py [hunk 1]
```diff
@@ -546,6 +546,40 @@ def load_density(self, fname):
...
+            # data[1]: comoving energy density [erg/cm^3]
+            eps = data[0].to_numpy()
+            n = data[1].to_numpy()
+            
+            self.data[z] = (eps * eV) , (n * erg * 1e6 / (eps * eV)**2)   # [J] , dn/deps [1/m^3/J]
+            
+            # adding the energies only once, cause they are the same in every file
+            if i == 0:
+                tmp_dict['energy'] = eps
+            tmp_dict[str(z)] = n
```
Comment (JulienDoerner) [inline photonField.py:577]: You can directly append to `self.photonDensity`. This would save a lot of overhead in storing the data.

**Review by JulienDoerner — CHANGES_REQUESTED** [review]
In the calculation of the Bete-Heitler pair production the following warning is raised: 

> IRB_Finke22
/mnt/e/UNI/CRPropa3-data/calc_pairproduction.py:64: IntegrationWarning: The maximum number of subdivisions (50) has been achieved.
  If increasing the limit yields no improvement it is advised to analyze
  the integrand in order to determine the difficulties.  If the position of a
  local difficulty can be determined (singularity, discontinuity) one will
  probably gain from splitting up the interval and calling the integrator
  on the subranges.  Perhaps a special-purpose integrator should be used.
  rate[i], err[i] = integrate.quad(

Did you test if there are any issues here?

**Comment by lukasmerten** [comment]
> In the calculation of the Bete-Heitler pair production the following warning is raised:
> [...]
> Did you test if there are any issues here?

This warning shows up for all photon fields. Since we never found any problems related to this, I would not fix it here.

File: photonField.py (hunk 1 above)
Comment (lukasmerten) [reply photonField.py]: I updated that part a bit. However, the tmp construction seems useful to me. The data is differently structured in the raw files compared to our storage. Therefore, when loading the files it somehow has to be transposed before stored into the photonField variable; or a more complex loading would be needed.

**Comment by lukasmerten** [comment]
@JulienDoerner are you ok with the proposed changes?

**Review by JulienDoerner — APPROVED** [review]
This looks fine for me. I guess the overhead in memory is only relevant for the initial phase. If there are issues at a later point we can still touch it then.

**Pull request merged**